### PR TITLE
Minor tweaks as I used it.

### PR DESCRIPTION
**Pull request by deads2k**
1. fixes message typo
2. creates namespace if the namespace isn't present
3. fixes a missing condition. conditions should always be set before return
4. tightens expression of condition setting to make it slightly easier to read
5. creates all deployments and then checks to see if those deployments are working. This should be in a separate controller eventually.

### CODE REVIEW

**Comment by deads2k** [comment]
/hold

this should cause some tests to fail.  If it doesn't, I'll have to add those tests.

**Comment by pmorie** [comment]
This LGTM, @qiujian16 can tag when he is ready.

/retest

File: pkg/operators/spoke/controller.go [hunk 1]
```diff
@@ -228,86 +229,84 @@ func (n *nucleusSpokeController) sync(ctx context.Context, controllerContext fac
...
-			})
-			helpers.UpdateNucleusSpokeStatus(
-				ctx, n.nucleusClient, spokeCoreName, helpers.UpdateNucleusSpokeConditionFn(spokeCore.Status.Conditions...))
+			helpers.UpdateNucleusSpokeStatus(ctx, n.nucleusClient, spokeCoreName, helpers.UpdateNucleusSpokeConditionFn(nucleusapiv1.StatusCondition{
+				Type: spokeRegistrationDegraded, Status: metav1.ConditionTrue, Reason: "ClusterNameMissing",
+				Message: fmt.Sprintf("Failed to get cluster name from `kubectl get secret -n %q %q -ojsonpath='{.data.cluster-name}`.  This is set by the spoke registration deployment.", hubSecret.Namespace, hubSecret.Name),
+			}))
 			return fmt.Errorf("Failed to get cluster name")
 		}
 		config.ClusterName = string(clusterName)
```
Comment (qiujian16) [inline pkg/operators/spoke/controller.go:292]: I think we should get cluster name from secret before deploying work agent. If cluster name is not specified in spokecore CR, we will not be able to update the deployment.

**Comment by qiujian16** [comment]
it is neat!

Only need to handle the case of cluster name is not set in spokecore. I will followup on that.

/lgtm

**Pull request merged**